### PR TITLE
Adding unit tests for AzureStackAdmin

### PR DESCRIPTION
**Pull request by bganapa**

### CODE REVIEW

File: src/ResourceManager/AzureStackAdmin/Commands.AzureStackAdmin.Test/Common/Utilities.psm1 [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+{
+    param([ScriptBlock] $ScriptBlock, [Object] $Argument, [int] $MaxTries, [int] $IntervalInSeconds)
+
+    if ($IntervalInSeconds -eq 0) { $IntervalInSeconds = 60  }
+
+    $result = Invoke-Command -ScriptBlock $ScriptBlock -ArgumentList $Argument;
+    $tries = 1;
+    while(( $result -ne $true) -and ($tries -le $MaxTries))
+    {
+        Start-Sleep -Seconds $IntervalInSeconds
```
Comment (hovsepm) [inline src/ResourceManager/AzureStackAdmin/Commands.AzureStackAdmin.Test/Common/Utilities.psm1]: please use Start-TestSleep instead

File: tools/Initialize-AzureInternalEnvironments.ps1 [hunk 2]
```diff
@@ -0,0 +1,26 @@
+$AzureStack="AzureStack"
```
Comment (hovsepm) [inline tools/Initialize-AzureInternalEnvironments.ps1]: Any reason to add this file here?

File: src/ResourceManager/AzureStackAdmin/Commands.AzureStackAdmin/Microsoft.Gallery/GalleryItems/AddGalleryItem.cs [hunk 3]
```diff
@@ -59,6 +60,19 @@ public class AddGalleryItem : AdminApiCmdlet
         [ValidateNotNull]
         public string Path { get; set; }
 
+
```
Comment (hovsepm) [inline src/ResourceManager/AzureStackAdmin/Commands.AzureStackAdmin/Microsoft.Gallery/GalleryItems/AddGalleryItem.cs]: remove extra spaces

File: src/ResourceManager/AzureStackAdmin/Commands.AzureStackAdmin/Microsoft.Gallery/GalleryItems/AddGalleryItem.cs [hunk 4]
```diff
@@ -59,6 +60,19 @@ public class AddGalleryItem : AdminApiCmdlet
...
+        /// every time the test runs
+        /// </summary>
+        public static Queue<Guid> GalleryPackageIds { get; set; }
+
+        static AddGalleryItem()
+        {
+            GalleryPackageIds = new Queue<Guid>();
+        }
+
+
```
Comment (hovsepm) [inline src/ResourceManager/AzureStackAdmin/Commands.AzureStackAdmin/Microsoft.Gallery/GalleryItems/AddGalleryItem.cs]: remove extra spaces

File: src/ResourceManager/AzureStackAdmin/Commands.AzureStackAdmin.Test/Common/Commands.AzureStackAdmin.Test.Common.csproj [hunk 5]
```diff
@@ -0,0 +1,152 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="..\..\..\packages\xunit.runner.visualstudio.2.1.0-beta4-build1109\build\net20\xunit.runner.visualstudio.props" Condition="Exists('..\..\..\packages\xunit.runner.visualstudio.2.1.0-beta4-build1109\build\net20\xunit.runner.visualstudio.props')" />
```
Comment (hovsepm) [inline src/ResourceManager/AzureStackAdmin/Commands.AzureStackAdmin.Test/Common/Commands.AzureStackAdmin.Test.Common.csproj]: xunit should be xunit.runner.visualstudio.2.1.0 version. Please look to other test projects in the ReousreManager folder.

Comment (hovsepm) [inline src/ResourceManager/AzureStackAdmin/Commands.AzureStackAdmin.Test/Common/Commands.AzureStackAdmin.Test.Common.csproj]: however it does not seems to be useful in this project. You can remove it completely.

File: src/ResourceManager/AzureStackAdmin/Commands.AzureStackAdmin.Test/Commands.AzureStackAdmin.Test.csproj [hunk 6]
```diff
@@ -0,0 +1,198 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="..\..\..\packages\xunit.runner.visualstudio.2.1.0-beta4-build1109\build\net20\xunit.runner.visualstudio.props" Condition="Exists('..\..\..\packages\xunit.runner.visualstudio.2.1.0-beta4-build1109\build\net20\xunit.runner.visualstudio.props')" />
```
Comment (hovsepm) [inline src/ResourceManager/AzureStackAdmin/Commands.AzureStackAdmin.Test/Commands.AzureStackAdmin.Test.csproj]: xunit should be xunit.runner.visualstudio.2.1.0 version. Please look to other test projects in the ReousreManager folder.

File: src/ResourceManager/AzureStackAdmin/Commands.AzureStackAdmin.Test/Commands.AzureStackAdmin.Test.csproj [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+    </Reference>
+    <Reference Include="Microsoft.Azure.Test.HttpRecorder">
+      <SpecificVersion>False</SpecificVersion>
+      <HintPath>..\..\..\packages\Microsoft.Azure.Test.HttpRecorder.1.0.5799.28345-prerelease\lib\net45\Microsoft.Azure.Test.HttpRecorder.dll</HintPath>
+    </Reference>
+    <Reference Include="Newtonsoft.Json">
+      <SpecificVersion>False</SpecificVersion>
+      <HintPath>..\..\..\packages\Newtonsoft.Json.6.0.8\lib\net45\Newtonsoft.Json.dll</HintPath>
+    </Reference>
+    <Reference Include="xunit, Version=1.9.2.1705, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c, processorArchitecture=MSIL">
```
Comment (hovsepm) [inline src/ResourceManager/AzureStackAdmin/Commands.AzureStackAdmin.Test/Commands.AzureStackAdmin.Test.csproj]: upgrade tp version 2.1.0

File: src/ResourceManager/AzureStackAdmin/Commands.AzureStackAdmin.Test/AzureStackTests.ps1 [hunk 8]
```diff
@@ -0,0 +1,619 @@
...
+{
+    $resourceGroupName = "GalleryItems"
+    $galleryItemName = "Microsoft.SimpleVMTemplate.1.0.0"
+    $galleryApiVersion = "2015-04-01"
+
+    try
+    {
+        New-AzureRmResourceGroup -Name $resourceGroupName  -Location redmond -Force
+		
+		[Microsoft.AzureStack.Commands.AddGalleryItem]::GalleryPackageIds.Enqueue("1988820c-2bcc-4682-9991-bec44e6b8324")
```
Comment (hovsepm) [inline src/ResourceManager/AzureStackAdmin/Commands.AzureStackAdmin.Test/AzureStackTests.ps1]: please align indentation in this file.

File: src/ResourceManager/AzureStackAdmin/Commands.AzureStackAdmin.Test/AzureStackTests.cs [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+        }
+
+        [Fact]
+        public void TestGalleryItem()
+        {
+            var testRunner = AzStackTestRunner.NewInstance;
+            testRunner.ApiVersion = "2015-04-01";
+            testRunner.RunPsTest("Test-GalleryItem");
+        }
+
```
Comment (hovsepm) [inline src/ResourceManager/AzureStackAdmin/Commands.AzureStackAdmin.Test/AzureStackTests.cs]: remove line extra break

File: src/ResourceManager/AzureStackAdmin/Commands.AzureStackAdmin.Test/Common/Utilities.psm1 (hunk 1 above)
Comment (bganapa) [inline src/ResourceManager/AzureStackAdmin/Commands.AzureStackAdmin.Test/Common/Utilities.psm1]: removing this Retry-Function from my utilities and using the Retry-Function from common.ps1

File: tools/Initialize-AzureInternalEnvironments.ps1 (hunk 2 above)
Comment (bganapa) [inline tools/Initialize-AzureInternalEnvironments.ps1]: removing this file from this repo

**Comment by bganapa** [comment]
http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/258/ - powershell sign

**Comment by hovsepm** [comment]
sign: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/269/

**Comment by hovsepm** [comment]
demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/532/

**Comment by bganapa** [comment]
sign - http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/
this is after merging with the latest dev as azure/dev branch had a signing failure for missing .snk file

**Pull request merged**